### PR TITLE
Prevent OutputParticle context from shifting while resizing

### PR DESCRIPTION
**Pull request by julienamsellem**
### Purpose of this PR
[1404749](https://fogbugz.unity3d.com/f/cases/1404749/)

Steps to repro:
- Open/Create a VFX Graph 
- Select the output particle context
- Resize the inspector panel
- You may need to unselect and reselect multiple times the context to make the issue happen

Unity_dSPVNPwxoJ
Unity_uzIFEqwPSH

---
### Testing status
Tested all contexts (with or without label) only the ParticleOutput context had this issue.

---
### Comments to reviewers
This issue does not seems new, I could reproduce it on master (2022.2) from November 29 2021.
According to the fix, the original issue could happen on any node which contains a **delayed int field**

### CODE REVIEW

**Review by VitaVFX — APPROVED** [review]
Thanks for the fix! Quickly double checked reproduction and several node combinations that has delayed field (Sort Priority, Get Attribute angle, Probability Sampling, Capacity + label in Init) with various sizes of Inspector. Issue is no longer reproducible.

**Comment by ludovic-theobald** [comment]
The current fix leads to the elements after Sorting Priority to disappear from the inspector (because we early-exit the function)

Before : 
image

After:
image

File: com.unity.visualeffectgraph/Editor/Inspector/VFXSlotContainerEditor.cs [hunk 1]
```diff
@@ -110,6 +110,8 @@ public virtual SerializedProperty DoInspectorGUI()
                     prop.Value.intValue = newValue;
                     modifiedSetting = prop.Value;
                 }
+
+                return modifiedSetting;
```
Comment (ludovic-theobald) [inline com.unity.visualeffectgraph/Editor/Inspector/VFXSlotContainerEditor.cs:114]: Early exit here leads to the rest of the settings to be ignored.

**Review by ludovic-theobald — CHANGES_REQUESTED** [review]

File: com.unity.visualeffectgraph/Editor/Inspector/VFXSlotContainerEditor.cs (hunk 1 above)
Comment (julienamsellem) [reply com.unity.visualeffectgraph/Editor/Inspector/VFXSlotContainerEditor.cs]: Really nice catch!
I didn't see that I added the early return in a for loop...

**Review by ludovic-theobald — APPROVED** [review]

**Comment by julienamsellem** [comment]
This PR has been migrated to new repo https://github.cds.internal.unity3d.com/unity/unity/pull/8749

**Comment by julienamsellem** [comment]
Backported to 2022.1 here https://github.cds.internal.unity3d.com/unity/unity/pull/11721

**Pull request closed without merging**